File: SISMUEBLUM JAVA/Venta.java
import java.util.Date;
import java.util.*;
public class Venta{
	private int id;
	private Date fechaCompra;
	private Cliente comprador;
	private TipoDocDeVenta tipoDoc;
	private List<DetalleVenta> cantPorProducto;
	private double subTotal;
	private double total;
	private Vendedor vendedor;
	private Date fechaEntrega;
	private String direccionDeDespacho;
	private TipoGradoSatisfaccion satisfaccion;
	
	public Venta(int id, Date fecha, Cliente comprador, TipoDocDeVenta tipoDoc, double subTotal, double total, Vendedor vendedor, Date fechaEntrega) {
		this.id = id;
		this.fechaCompra = fechaCompra;
		this.comprador = comprador;
		this.tipoDoc = tipoDoc;
		this.subTotal = subTotal;
		this.total = total;
		this.vendedor = vendedor;
		this.fechaEntrega = fechaEntrega;
		cantPorProducto = new ArrayList<DetalleVenta>();
	}
	
	public Venta(int id, Date fecha, Cliente comprador, TipoDocDeVenta tipoDoc, double subTotal, double total, Vendedor vendedor, Date fechaEntrega, String direccionDeDespacho) {
		this.id = id;
		this.fechaCompra = fechaCompra;
		this.comprador = comprador;
		this.tipoDoc = tipoDoc;
		this.subTotal = subTotal;
		this.total = total;
		this.vendedor = vendedor;
		this.fechaEntrega = fechaEntrega;
		this.direccionDeDespacho = direccionDeDespacho;
		cantPorProducto = new ArrayList<DetalleVenta>();
	}
	
	public int getId () {return this.id;}
	public void setId(int valor) {this.id = valor;}
	public Date getFechaCompra () {return this.fechaCompra;}
	public void setFechaCompra (Date valor) {this.fechaCompra = valor;}
	public Cliente getCliente () {return this.comprador;}
	public void setCliente(Cliente valor) {this.comprador = valor;}
	public TipoDocDeVenta getTipoDoc () {return this.tipoDoc;}
	public void setTipoDoc(TipoDocDeVenta valor) {this.tipoDoc = valor;}
	public double getSubTotal () {return this.subTotal;}
	public void setSubTotal(double valor) {this.subTotal = valor;}
	public double getTotal () {return this.total;}
	public void setTotal(double valor) {this.total = valor;}
	public Vendedor getVendedor () {return this.vendedor;}
	public void setVendedor (Vendedor valor) {this.vendedor = valor;}
	public Date getFechaEntrega () {return this.fechaEntrega;}
	public void setFechaEntrega (Date valor) {this.fechaEntrega = valor;}
	public String getDireccionDeDespacho () {return this.direccionDeDespacho;}
	public void setDireccionDeDespacho (String valor) {this.direccionDeDespacho = valor;}
	public TipoGradoSatisfaccion getSatisfaccion () {return this.satisfaccion;}
	public void setSatisfaccion(TipoGradoSatisfaccion valor){this.satisfaccion = valor;}
	
	public void añadirProducto(Producto producto, double cantidad, double descuento){
		DetalleVenta item;
		if (descuento>0) item = new DetalleVenta(producto, cantidad, descuento);
		else item = new DetalleVenta(producto, cantidad);
		
		cantPorProducto.add(item);
	}
	
	public String consultarVenta(){return "";}
}

File: SISMUEBLUM C#/Venta.cs
using System;
public class Venta{
	private int _id;
	private DateTime _fechaCompra;
	private Cliente _comprador;
	private TipoDocDeVenta _tipoDoc;
	private List<DetalleVenta> _cantPorProducto;
	private double _subTotal;
	private double _total;
	private Vendedor _vendedor;
	private DateTime _fechaEntrega;
	private string _direccionDeDespacho;
	private TipoGradoSatisfaccion _satisfaccion;
	
	public Venta(int id, DateTime fecha, Cliente comprador, TipoDocDeVenta tipoDoc, double subTotal, double total, Vendedor vendedor, DateTime fechaEntrega) {
		this._id = id;
		this._fechaCompra = fechaCompra;
		this._comprador = comprador;
		this._tipoDoc = tipoDoc;
		this._subTotal = subTotal;
		this._total = total;
		this._vendedor = vendedor;
		this._fechaEntrega = fechaEntrega;
		_cantPorProducto = new List<DetalleVenta>();
	}
	
	public Venta(int id, DateTime fecha, Cliente comprador, TipoDocDeVenta tipoDoc, double subTotal, double total, Vendedor vendedor, DateTime fechaEntrega, string direccionDeDespacho) {
		this._id = id;
		this._fechaCompra = fechaCompra;
		this._comprador = comprador;
		this._tipoDoc = tipoDoc;
		this._subTotal = subTotal;
		this._total = total;
		this._vendedor = vendedor;
		this._fechaEntrega = fechaEntrega;
		this._direccionDeDespacho = direccionDeDespacho;
		_cantPorProducto = new List<DetalleVenta>();
	}
	
	~Venta(){}
	
	public int id{get{return this._id;} set{this._id = value;}}
	public DateTime fechaCompra{get{return this._fechaCompra;} set{this._fechaCompra = value;}}
	public Cliente comprador{get{return this._comprador;} set{this._comprador = value;}}
	public TipoDocDeVenta tipoDoc{get{return this._tipoDoc;} set{this._tipoDoc = value;}}
	public double subTotal{get{return this._subTotal;} set{this._subTotal = value;}}
	public double total{get{return this._total;} set{this._total = value;}}
	public Vendedor vendedor{get{return this._vendedor;} set{this._vendedor = value;}}
	public DateTime fechaEntrega{get{return this._fechaEntrega;} set{this._fechaEntrega = value;}}
	public string direccionDeDespacho{get{return this._direccionDeDespacho;} set{this._direccionDeDespacho = value;}}
	public TipoGradoSatisfaccion satisfaccion{get{return this._satisfaccion;} set{this._satisfaccion = value;}}
	
	public void añadirProducto(Producto producto, double cantidad, double descuento){
		DetalleVenta item;
		if (descuento>0) item = new DetalleVenta(producto, cantidad, descuento);
		else item = new DetalleVenta(producto, cantidad);
		
		_cantPorProducto.Add(item);
	}
	
	public string consultarVenta(){return "";}
}